### PR TITLE
added support to RFCs 2518 & 4918 WEBDAV METHODS

### LINKED ISSUE #1426: WebDAV methods not supported
Falcon defines supported HTTP methods in `falcon/constants.py`: supported are "usual" `HTTP_METHODS` and, in addition to that, `WEBDAV_METHODS`. However, only WebDAV versioning extension methods from RFC 3253 are supported, but not the "ordinary" WebDAV ones (i.e. from RFCs 2518 & 4918) like `COPY`, `LOCK`, `MKCOL`, `MOVE` etc.

Supporting only an extension, but not the core upon which that extension builds looks somewhat inconsistent.

### PR DESCRIPTION
**Pull request by screamingskulls**
Fixes #1426

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -431,6 +433,7 @@ Many thanks to all of our talented and stylish contributors to this release!
 - `steven-upside <https://github.com/steven-upside>`_
 - `tribals <https://github.com/tribals>`_
 - `vytas7 <https://github.com/vytas7>`_
+- `screamingskulls <https://github.com/screamingskulls>`_
```
Comment (vytas7) [inline CHANGES.rst]: This list was specific to Falcon 2.0, we should create a new one under 3.0.

File: CHANGES.rst [hunk 2]
```diff
@@ -389,6 +389,8 @@ Fixed
   compliance with RFC 7232.
 - Fixed ``TypeError`` being raised when using Falcon's testing framework
   to simulate a request to a generator-based WSGI app.
+- Falcon now supports all WebDAV methods (RFC 2518 and RFC 4918), such as
```
Comment (vytas7) [inline CHANGES.rst]: * We should move this note to the 3.0.0 section. 2.0 has already been released.
* We should also add this note to `3.0.0` RST file.

File: falcon/constants.py [hunk 3]
```diff
@@ -11,19 +11,26 @@
...
     'REPORT',
     'UNCHECKIN',
+    'UNLOCK'
     'UPDATE',
     'VERSION-CONTROL',
 )
 
 COMBINED_METHODS = HTTP_METHODS + WEBDAV_METHODS
 
-
```
Comment (vytas7) [inline falcon/constants.py]: Any specific reason for re-styling whitespace here?
(Or just trying to make a compelling case for `black` :wink: )

File: CHANGES.rst (hunk 2 above)
Comment (nZac) [reply CHANGES.rst]: Actually, we need a new `docs/_newsfragment/` for this. I would think this should be classified as a bug fix.

Comment (vytas7) [reply CHANGES.rst]: Now with the Towncrier setup, yes :slightly_smiling_face: 
Agreed wrt bug fix.

**Comment by vytas7** [comment]
Closing this as it has been superseded by https://github.com/falconry/falcon/pull/1588

**Pull request closed without merging**